### ISSUE TITLE
Ability to turn link preview off per channel

### ISSUE THREAD

**Issue by pahakalle** [thelounge/thelounge#4805]
### Feature Description
Would be nice to be able to turn off link preview on some channels, but still have it on others.
Could maybe be as an option bellow the mute channel.

**Comment by ThatLurker** [comment]
The main issues arises when multiple thelounge users make requests to a link which might have a rate limited or just a bad server and in doing so they "ddos" the website.
A per channel toggle would help in this case as you would still be able to use link preview on other smaller channels instead of the all or nothing you can currently do in the configs.

**Comment by brunnre8** [comment]
If you have a server that can't handle a few irc users, you're gonna have a bad time on the internet.

Considering that the default of TL is to not show previews, people explicitly turned it on, probably because they want to have it.

Meaning your server will still get ddosed, by your measure, as those will keep it enabled.

If you really care, you can filter us on the server, we identify as twitterbot so it's easy enough

**Comment by TheTechRobo** [comment]
It's still a problem when someone sends a link to a server that happens to have this issue, and suddenly nobody can access it anymore.

**Comment by ThatLurker** [comment]
Yeah, I totally agree that that should be the case and this is not because of my servers sadly.
But still, adding a toggle for it per channel would make it even easier for people to toggle the feature on or off on channels that don't benefit from it. But still, leave it enabled on, for example, more media/meme heavy channels, as those types of stuff can easily take the traffic. But again, of course, this would require the user to turn it off themselves, but at least now they can still use it on other channels.

**Comment by JustAnotherArchivist** [comment]
I'm one of the operators of the channel that triggered the creation of this issue, and I'd like to give a bit more context why this is needed.

We have a large number of channels, but one single channel receives a lot more messages than the others, including a lot of automated activity. On an average day, that'll be several thousand messages, over half of which feature a URL. By nature of our activities, this channel also receives a higher proportion of links to servers with a very poor performance. We do not control the servers in question, and we cannot avoid linking to them. The Lounge users with link previews enabled are, with some regularity, overwhelming such servers solely because a link appeared in this channel. Due to the high traffic in the channel, the value of previews in there is also minimal, and it's virtually guaranteed that most users will never see the vast majority of the previews.

I would greatly prefer being able to point The Lounge users at a configuration option to disable previews for that specific channel over the only viable alternative: forcing them to disable previews globally to avoid getting kickbanned, which is unfortunately the policy we now have to adopt until the situation changes.

**Comment by duckspike** [comment]
I'm in support of this feature proposal. This might be a niche feature to us, but it's an important feature to have as there may be other niche instances where a user is in a channel that they may not want inline previews turned on for (i.e. an NSFW channel while they're at work, or a channel where lots of images or other content gets posted).

**Comment by brunnre8** [comment]
as always, refrain from me too style comments, they do not help the discussion in any way.
Use thumbs up reactions if you feel like it adds value.
